### ISSUE TITLE
Accessibility Issue: Keyboard focus for interactive elements inside Tooltip

### ISSUE THREAD

**Issue by alexandreabeh** [uber/baseweb#4557]
Navigating to a tooltip via keyboard will display the tooltip but not autoFocus on the first interactive element. 

Repro: go to this section for [Tooltip docs](https://baseweb.design/components/tooltip/#tooltip-stateful-example-with-complex-content) and hit tab once. This will open the 'complex content' tooltip which contains a link. The link is not focused, and cannot be clicked via keyboard navigation in any way.

## Current Behavior

Keyboard navigation skips the interactive element inside the tooltip. In Base docs, this can be seen in the example that contains a link. In my development environment, I have observed this behavior with buttons and inputs as well.

## Expected Behavior

Keyboard navigation includes navigating to the interactive element.

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v10.1.1  |
| React   | 16.14 |
| browser |  Chrome 94 |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

Similar to https://github.com/uber/baseweb/issues/3840, but that issue and fix did not cover accessibility via keyboard navigation.
